program/src/processor.rs

```diff
@@ -2012,21 +2057,64 @@ impl Processor {
         };
 
         // Enforce order price limits if the order is a limit order that goes on the book
-        let native_price = {
-            // Conver the price in
+        let (coin_lot_size, pc_lot_size) = {
             let market = load_market_state(spot_market_ai, dex_prog_ai.key)?;
-            I80F48::from_num(order.limit_price.get())
-                .checked_mul(I80F48::from_num(market.pc_lot_size))
-                .unwrap()
-                .checked_div(I80F48::from_num(market.coin_lot_size))
-                .unwrap()
+            (market.coin_lot_size, market.pc_lot_size)
         };
+        let native_price = I80F48::from_num(order.limit_price.get())
+            .checked_mul(I80F48::from_num(pc_lot_size))
+            .unwrap()
+            .checked_div(I80F48::from_num(coin_lot_size))
+            .unwrap();
+
         let oracle_price = mango_cache.get_price(market_index);
         let info = &mango_group.spot_markets[market_index];
         let market_open_orders_ai = open_orders_ais[market_index].unwrap();
 
         // If not post_allowed, then pre_locked may not increase
-        let (post_allowed, pre_locked) = {
+        let is_luna_market = spot_market_ai.key == &luna_spot_market::ID;
+        let (post_allowed, pre_locked) = if is_luna_market {
+            let open_orders = load_open_orders(market_open_orders_ai)?;
+            // only one open order at a time
+            check!(
+                open_orders.free_slot_bits == u128::MAX,
+                MangoErrorCode::InvalidOrderInClosingMarket
+            )?;
+
+            // For markets in close only mode:
+            // if long, sell quantity may only be as large as position
+            // if short, buy quantity may only be as large as position
+            let net_pos = mango_account
+                .get_net(&mango_cache.root_bank_cache[market_index], market_index)
+                .checked_add(I80F48::from_num(open_orders.native_coin_total))
+                .unwrap();
+
+            let order_quantity = order.max_coin_qty.get().checked_mul(coin_lot_size).unwrap();
+            match order_side {
+                serum_dex::matching::Side::Bid => {
+                    check!(
+                        net_pos.is_negative() && order_quantity <= net_pos.abs().to_num::<u64>(),
+                        MangoErrorCode::InvalidOrderInClosingMarket
+                    )?;
+
+                    (
+                        order.limit_price.get() < 10,
```

Comment:
maybe we could compare `native_price` instead, then we don't need to double check coin_lots and pc_lots to see what the constant means.

Reply:
isn't native price also confusing?